sqlglot/optimizer/merge_subqueries.py

```diff
@@ -119,6 +119,19 @@ def _mergeable(outer_scope, inner_select, leave_tables_isolated, from_or_join):
     Returns:
         bool: True if can be merged
     """
+
+    def _get_window_expressions_in_where_clause():
+        window_expressions = list(inner_select.find_all(exp.Window))
+        window_alias_names = [window.parent.alias_or_name for window in window_expressions]
+        inner_select_name = inner_select.parent.alias_or_name
+        where_clauses_columns = [column for column in outer_scope.columns if column.find_ancestor(exp.Where)]
```

Comment:
you're checking every column is in a where? why not find the where first and then find the columns?

Reply:
Actually I'm wondering if I should instead check if the column is used anywhere besides a select expression. That would cover other cases like a window in a group by or join.

Reply:
Yeah I think it's allowed in some clauses (selections... maybe order by?) but not others.

Reply:
Yeah I tested a few cases and explicitly excluded them in my latest commit (also updated the tests).